tests/unit/mathematics/linalg/operations/Eigen3_matrix_operations_unittest.cc

```diff
@@ -0,0 +1,31 @@
+#include <shogun/lib/config.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespaceMatrix.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+using namespace linalg;
+
+template <typename T>
+class LinalgBackendEigenMatrix : public ::testing::Test {};
+
+typedef ::testing::Types<float32_t, float64_t, floatmax_t> PTypes;
```

Comment:
yeah nice to do a typed test.
(offtopic: this should be done in lots of other places in shogun tests as well)